### ISSUE TITLE
Teardown conflict with task scheduler

### ISSUE THREAD

**Issue by vintagesucks** [mehrancodes/laravel-harbor#106]
Hi! I noticed that my "Sites" navigation on Forge contained some sites that should have been deleted already.

![grafik](https://github.com/mehrancodes/laravel-harbor/assets/13335308/26fd8352-f209-4ac0-a6fa-69d785683d0a)

In fact, the site overview for that server no longer even shows the site, indicating that _something_ has happened:

![grafik](https://github.com/mehrancodes/laravel-harbor/assets/13335308/07d82959-165e-43a8-a3fd-0f9aeac3c086)

This happens on every teardown, but only when I include `FORGE_JOB_SCHEDULER=true` in my environment.

To narrow it down, I reduced the pipeline in [TearDownCommand](https://github.com/mehrancodes/laravel-harbor/blob/8fcba4a9af31b073d724b327d56513e5272a58f5/app/Commands/TearDownCommand.php) to this:

```php
->through([
  FindServer::class,
  FindSiteOrFail::class,
  RemoveTaskScheduler::class,
  DestroySite::class,
])
```

Only when `RemoveTaskScheduler` is included here does the above bug occur. I _think_ Forge is silently failing to actually remove the site if the scheduled job is still being removed. If I add something like `sleep(10)` before `DestroySite`, everything is deleted as expected:

```php
->through([
  FindServer::class,
  FindSiteOrFail::class,
  RemoveTaskScheduler::class,
  function (ForgeService $service, Closure $next) {
    sleep(10);
    return $next($service);
  },
  DestroySite::class,
])
```

I am not sure if this is a bug in the Forge API/SDK or if Harbor is somehow supposed to wait for the job to actually be deleted, but this is currently resulting in many orphaned sites on my Forge server.

Complete environment from my local test:

```env
SUBDOMAIN_NAME="test"
FORGE_DOMAIN="<removed>"
FORGE_GIT_BRANCH="main"
FORGE_GIT_REPOSITORY="<removed>"
FORGE_JOB_SCHEDULER=true
FORGE_PHP_VERSION="php82"
FORGE_SERVER="<removed>"
FORGE_TOKEN="<removed>"
FORGE_DEPLOY_SCRIPT="cd $FORGE_SITE_PATH"
```

Command I can use to reproduce the problem locally:

```sh
php ./harbor provision && php ./harbor teardown
```

This will return `Environment teardown successful! All provisioned resources have been removed.` and the site is partially removed from some parts of Forge, but remains in the site navigation and, more importantly, on the actual server.

**Comment by shawnhooper** [comment]
I've had this same problem.  I haven't yet dug as deeply as @vintagesucks has into it.  I have had to login to the server a few times and delete sites manually to clear up space.  Sometimes in the file system directly, as they were already removed from the Sites dashboard. 

Maybe related, I've also seen my GitHub account run of out deployment keys.

**Comment by Hannsre** [comment]
I've come across this as well. A new PR Site will be spawned without issues usually, but on teardown it won't be removed completely. 
Haven't dug as deep yet as well, but I can't see any remnants of the preview Site in the forge Dashboard. But if I check via the Forge API, the removed site is still there in Status `removing`:

```
    "site": {
        "id": (removed),
        "server_id": (removed),
        "name": "pr-Id.example.com",
        "aliases": [],
        "directory": "/public",
        "wildcards": false,
        "status": "removing",
  ```

I can delete this site using the forge API manually then, but this shouldn't be necessary. 

I can also confirm that setting `FORGE_JOB_SCHEDULER` to `False` will solve this issue for me.

**Comment by samshupac** [comment]
I'm also experiencing this same issue.
This solution of setting `FORGE_JOB_SCHEDULER: false` isn't ideal as it means the temporary environments we create can't be used to test features fully.

FYI @mehrancodes

**Comment by RealMrHex** [comment]
Hey @vintagesucks, thanks for your report and explanations.
I'm working on this issue and ill update u all asap.

**Comment by samshupac** [comment]
Hey @RealMrHex / @mehrancodes - is there possibly an update on this? thanks

**Comment by mehrancodes** [comment]
I ran into an issue while testing a feature for Harbor: if we skip the **RemoveTaskScheduler**, the site just sits in a **removing** status. After running the teardown command a couple of times, it finally deletes.

I’m planning to bring **RemoveTaskScheduler** back into the teardown pipeline and add a short delay on it, so it removes the scheduler before we try deleting the Forge site.

@samshupac, @vintagesucks, would you mind testing this with your setup once the PR is ready?

**Comment by vintagesucks** [comment]
@mehrancodes sure 🚀

**Comment by mehrancodes** [comment]
@vintagesucks, feel free to close this issue if it did work for you:)
